packages/blockprotocol/cli/commands/publish/post-form.js

```diff
@@ -35,33 +37,46 @@ export const postPublishForm = async ({
 
   // @todo when we add another command that connects to the API this should be moved to a shared location
   return fetch(url, options)
-    .then(async (resp) => {
+    .then(async (response) => {
       try {
-        return await resp.json();
-      } catch (err) {
+        return /** @type {{ block: { blockSitePath: string }}} */ (
+          await response.json()
+        );
+      } catch (error) {
         // if the JSON parsing failed, the user is probably trying to contact a host that doesn't implement the endpoint
-        const errorMsg =
-          resp.status !== 200
-            ? `Request to ${blockProtocolSiteHost} errored: ${resp.status} ${resp.statusText}`
-            : `Malformed JSON response: ${err.message}`; // a non-parseable 200 status response should not happen
+        const errorMessage =
+          response.status !== 200
+            ? `Request to ${blockProtocolSiteHost} errored: ${response.status} ${response.statusText}`
+            : `Malformed JSON response: ${
+                error instanceof Error ? error.message : error
+              }`; // a non-parsable 200 status response should not happen
         return {
           errors: [
             {
-              msg: errorMsg,
+              msg: errorMessage,
             },
           ],
         };
       }
     })
-    .catch((err) => {
-      if (err.code === "ENOTFOUND" || err.code === "ECONNREFUSED") {
-        // the host is not contactable at all – probably a network issue or non-running dev/staging instance
-        return {
-          errors: [
-            { msg: `Could not connect to host ${blockProtocolSiteHost}` },
-          ],
-        };
+    .catch((/** @type unknown */ error) => {
+      if (typeof error === "object" && error) {
+        // @ts-expect-error -- code can be a property on the error object
+        const errorCode = error.code;
+        if (errorCode === "ENOTFOUND" || errorCode === "ECONNREFUSED") {
+          // the host is not contactable at all – probably a network issue or non-running dev/staging instance
+          return {
+            errors: [
+              { msg: `Could not connect to host ${blockProtocolSiteHost}` },
+            ],
+          };
+        }
+        if ("errors" in error) {
+          // should be a meaningful error provided by the API about why the request failed, e.g. bad API key
+          return error;
+        }
       }
-      return err; // should be a meaningful error provided by the API about why the request failed, e.g. bad API key
```

Comment:
Am I right that we expect something like this here? `{ errors: [{ msg: "hello" }] }`

Reply:
yes, exactly